src/Application/Actions/Hooks/StripeUpdate.php

```diff
@@ -36,18 +36,19 @@ public function __construct(
         $this->entityManager = $entityManager;
         $this->serializer = $serializer;
         // As `settings` is just an array for now, I think we have to inject Container to do this.
-        $this->webhookSecret = $container->get('settings')['stripe']['webhookSecret'];
+        $this->webhookSecret = (string) $container->get('settings')['stripe']['webhookSecret'];
 
         parent::__construct($logger);
     }
 
     protected function action(): Response
     {
+        $webhook = new StripeWebhook();
```

Comment:
The idea behind extending Stripe's class was to allow it be injected as a service and therefore mocked. If we are instantiating it directly here then tests are not mocking it.

Using the Di approach, we would instead take `StripeWebhook` as a constructor argument and then use the DI setup (as in `dependencies.php`) to tell the container how to provide that class.

However, given how far you've got without doing this, maybe my premise was wrong and we can cover everything without mocking Stripe's client here, instead testing with fake data but real Stripe client code. More on my other comments.